app/models/news_article.rb

```diff
@@ -4,11 +4,33 @@ class NewsArticle < Node
            source_type: 'Section'
 
   alias_method :publisher, :section
-
   store_attribute :data, :release_date, Date
 
-  validates_presence_of :section, :parent
-  validates :parent, ancestry_depth: { minimum: 2 }
+  friendly_id :date_and_name, use: [:slugged, :scoped], routes: :default, scope: :section
+
+  validates_presence_of :section
 
   scope :published, -> { where state: 'published' }
+
+  def layout
+    'news_article'
+  end
+
+  # News Articles require specific redirects to a custom route that
+  # is not defined by node hierarchy. As such, this method is a convenience
+  # wrapper around the url helpers, to allow this model to override the default
+  # redirect action.
+  def full_path
```

Comment:
But then you would be calling `nodes_path(node.path)` when technically a news article isn't using that route. Using `#full_path` here also means we no longer need to remember what the helper is called or what it's parameters are. And if we need to change it in the future it's all in one place (like when we had to remove the section from every call to nodes_path).